### PR TITLE
add e2e test for simple streaming server

### LINKED ISSUE #67: end-to-end tests
Add end-to-end tests for:

- [x] **dynamic Batching** - addressed by #68 
- [x] **Dynamic batching with streaming** - addressed by #68 
- [x] single prediction #70
- [x] single streaming #247

### PR DESCRIPTION
**Pull request by AdolfoVillalobos**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes #67:

- Added end-to-end test for the simple streaming server case, which was missing.

I hope I understood the issue correctly.! I'm entirely open to suggestions. 

## Did you have fun?

Yess. I hope I can contribute to this cool project :)

### CODE REVIEW

File: tests/e2e/test_e2e.py [hunk 1]
```diff
@@ -295,3 +295,24 @@ def test_openai_parity_with_response_format():
...
+def test_e2e_single_streaming():
+    resp = requests.post("http://127.0.0.1:8000/predict", json={"input": 4.0}, headers=None, stream=True)
+    assert resp.status_code == 200, f"Expected response to be 200 but got {resp.status_code}"
+
+    outputs = []
+    for line in resp.iter_lines():
+        if line:
+            outputs.append(json.loads(line.decode("utf-8")))
+
+    assert len(outputs) > 1, "Expected multiple streamed outputs"
```
Comment (aniketmaurya) [inline tests/e2e/test_e2e.py:310]:
```suggestion
```

File: tests/e2e/test_e2e.py [hunk 2]
```diff
@@ -295,3 +295,24 @@ def test_openai_parity_with_response_format():
...
+    assert len(outputs) == 3, "Expected 3 streamed outputs"
+    assert outputs[-1] == {"output": 12.0}, "Final output doesn't match expected value"
+
+    # Check that the outputs are incrementally building up to the final result
+    for i, output in enumerate(outputs[:-1]):
+        assert 0 < output["output"] <= 12.0, f"Intermediate output {i} is not within expected range"
+        if i > 0:
+            assert output["output"] >= outputs[i - 1]["output"], "Outputs are not monotonically increasing"
```
Comment (aniketmaurya) [inline tests/e2e/test_e2e.py:318]: This can be simplified. Just iterate through the outputs and assert the actual values [4, 8, 12]

**Review by aniketmaurya — APPROVED** [review]
Awesome, looks great @AdolfoVillalobos!! Added few comments.

File: tests/e2e/test_e2e.py (hunk 1 above)
Comment (aniketmaurya) [reply tests/e2e/test_e2e.py:310]: This is redundant since you already assert length == 3

Comment (AdolfoVillalobos) [reply tests/e2e/test_e2e.py:310]: I agree. Will remove!

File: tests/e2e/test_e2e.py (hunk 2 above)
Comment (AdolfoVillalobos) [reply tests/e2e/test_e2e.py]: Makes sense. I recently pushed a simplified version, asserting that the received outputs correspond to the elements  in the array [4, 8, 12]

**Pull request merged**